libsolidity/experimental/ast/TypeSystem.cpp

```diff
@@ -199,6 +200,82 @@ FixedTypeVariable TypeSystem::freshFixedTypeVariable(Sort _sort)
 	return freshFixedVariable(_sort);
 }
 
+experimental::Type TypeSystem::fixTypeVariables(Type const& _type)
```

Comment:
I don't think this will work like done here - we will have to use unification as base mechanism here, i.e. recursively traverse the type and whenever we see a generic type variable create a new fixed one and unify (if you continue to resolve properly, unification will actually make the ``genericToFixed`` map obsolete...

If we *don't* do that, the function body during code generation will have generic type variables for which we cannot tell what they mean in relation to the function type.

Reply:
For the reverse this will also be a bit of a headache - for that we'll need to use the concrete monomorphic function type and construct a substitution from fixed variables to instances to be used for codegen - but we can cross that bridge when we come to it, I expect this will mean a new kind of type environment that allows also tracking substitutions of fixed type variables not only generic ones to be used only in codegen (respectively for monomorphization)